File: ios/RNSystemSoundsModule.swift
import AudioToolbox
import ExpoModulesCore

public final class RNSystemSoundsModule: Module {
  public func definition() -> ModuleDefinition {
    Name("RNSystemSounds")

    Function("playSystemSound") { (soundID: Int) in
      AudioServicesPlaySystemSound(SystemSoundID(soundID))
    }
  }
}

File: android/src/main/java/expo/modules/rnsystemsounds/RNSystemSoundsModule.kt
package expo.modules.rnsystemsounds

import android.media.AudioManager
import android.media.ToneGenerator
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class RNSystemSoundsModule : Module() {
  private val toneGenerator = ToneGenerator(AudioManager.STREAM_SYSTEM, 100)

  override fun definition() = ModuleDefinition {
    Name("RNSystemSounds")

    Function("playSystemSound") { soundID: Int ->
      toneGenerator.startTone(soundID)
    }

    Function("stopSystemSound") {
      toneGenerator.stopTone()
    }

    OnDestroy {
      toneGenerator.release()
    }
  }
}
